pkgs/stdenv/darwin: bootstrap Darwin using update tools

reckenrode opened the pull request:
* Updates bootstrap tools to latest fresh tools (includes LLVM 19 and ld64 954.16); and
* Drops bootstrap tools workarounds from stdenv bootstrap, SDK, and ld64 builds.

Tools from:

* https://hydra.nixos.org/build/290554342
* https://hydra.nixos.org/build/290554343

Testing:

    maintainers/scripts/bootstrap-files/refresh-tarballs.bash --targets=aarch64-apple-darwin,x86_64-apple-darwin
    for sys in aarch64 x86_64; nix-build -A stdenv --system ${sys}-darwin

### aarch64-apple-darwin

sha256sum of files to be uploaded:

    $ sha256sum /nix/store/8wbnh6accpx3fcf20jvbnapcalvm1cmd-stdenv-bootstrap-tools/on-server/*
    8be2b04c863a3f774a7cd984e360560bdfaa5289a80f16b33b9d2ebebccd9691  /nix/store/8wbnh6accpx3fcf20jvbnapcalvm1cmd-stdenv-bootstrap-tools/on-server/bootstrap-tools.tar.xz
    4e61c00b561d859c5dacb498291c2509d3644536fd316b451c973b59ea4eff50  /nix/store/8wbnh6accpx3fcf20jvbnapcalvm1cmd-stdenv-bootstrap-tools/on-server/unpack.nar.xz

Suggested commands to upload files to 'tarballs.nixos.org':

    $ nix-store --realize /nix/store/8wbnh6accpx3fcf20jvbnapcalvm1cmd-stdenv-bootstrap-tools
    $ aws s3 cp --recursive --acl public-read /nix/store/8wbnh6accpx3fcf20jvbnapcalvm1cmd-stdenv-bootstrap-tools/on-server/ s3://nixpkgs-tarballs/stdenv/aarch64-apple-darwin/d8657587c0c1ce2191b4b489299c386b5b7148c6
    $ aws s3 cp --recursive s3://nixpkgs-tarballs/stdenv/aarch64-apple-darwin/d8657587c0c1ce2191b4b489299c386b5b7148c6 ./
    $ sha256sum bootstrap-tools.tar.xz unpack.nar.xz
    $ sha256sum /nix/store/8wbnh6accpx3fcf20jvbnapcalvm1cmd-stdenv-bootstrap-tools/on-server/*

### x86_64-apple-darwin

sha256sum of files to be uploaded:

    $ sha256sum /nix/store/y617yisg19x2pj34hzxg9zcm3wlsfj29-stdenv-bootstrap-tools/on-server/*
    82d91e694695f195d8f09c4493277b65e07be4c112a6c04f8dabb1d01d9cd251  /nix/store/y617yisg19x2pj34hzxg9zcm3wlsfj29-stdenv-bootstrap-tools/on-server/bootstrap-tools.tar.xz
    42d3cb8803a69673ea1ddd78de73eb63239825226c96c56ab170fe25d4ca79e3  /nix/store/y617yisg19x2pj34hzxg9zcm3wlsfj29-stdenv-bootstrap-tools/on-server/unpack.nar.xz

Suggested commands to upload files to 'tarballs.nixos.org':

    $ nix-store --realize /nix/store/y617yisg19x2pj34hzxg9zcm3wlsfj29-stdenv-bootstrap-tools
    $ aws s3 cp --recursive --acl public-read /nix/store/y617yisg19x2pj34hzxg9zcm3wlsfj29-stdenv-bootstrap-tools/on-server/ s3://nixpkgs-tarballs/stdenv/x86_64-apple-darwin/d8657587c0c1ce2191b4b489299c386b5b7148c6
    $ aws s3 cp --recursive s3://nixpkgs-tarballs/stdenv/x86_64-apple-darwin/d8657587c0c1ce2191b4b489299c386b5b7148c6 ./
    $ sha256sum bootstrap-tools.tar.xz unpack.nar.xz
    $ sha256sum /nix/store/y617yisg19x2pj34hzxg9zcm3wlsfj29-stdenv-bootstrap-tools/on-server/*

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

reckenrode committed:
pkgs/stdenv/darwin: update bootstrap tools

Updates bootstrap tools to latest fresh tools (includes LLVM 19 and ld64 954.16).

reckenrode committed:
pkgs/stdenv/darwin: remove old bootstrap tools workarounds

* The versions of cctools and ld64 in the bootstrap tools now support
  code-signing, so rely on them to do that instead of using a post-link
  hook that can sometimes fail to sign binaries;
* Drop the manual SDK header and library paths now that `clang` and
  `ld64` in the bootstrap tools correctly handle sysroots on Darwin.

reckenrode committed:
pkgs/stdenv/darwin: link `llvm-readtapi` in stage 0

reckenrode committed:
pkgs/stdenv/darwin: simplify SDK version expression

reckenrode committed:
pkgs/stdenv/darwin: add jq to stage 0

reckenrode committed:
ld64: remove old bootstrap tools workaround

Now that the `clang` in the bootstrap tools supports C23 `#embed`, the
patch can be applied unconditionally.

reckenrode committed:
apple-sdk: remove old bootstrap tools workaround

The stubs can unconditionally be processed now that the bootstrap tools
have both `jq` and `llvm-readtapi`.

reckenrode committed:
pkgs/stdenv/darwin: drop llvm-manpages from bootstrap

Building the LLVM man pages requires a bunch of Python dependencies,
which are annoying to build during the bootstrap.

reckenrode committed:
darwin.copyfile: remove old bootstrap tools workaround

The version of `clang` in the bootstrap tools now supports C23 empty
initializers.

reckenrode committed:
pkgs/stdenv/darwin: drop openbsm from the bootstrap

It’s not actually used by anything in the bootstrap, so avoid building
it unnecessarily.

emilazy reviewed (approved):
Yay!

One nit: we’re still overlaying `postLinkSignHook` in the bootstrap. We could presumably just drop the package entirely?

cc @NixOS/infra for bootstrap tools upload

paparodeo reviewed (approved):
LGTM: verified bootstrap tools hashes

reckenrode committed:
darwin.postLinkSignHook: drop and add to darwin-aliases.nix

The ld64 in the updated bootstrap tools is capable of code-signing, so
this hook is no longer needed.

reckenrode commented:
> One nit: we’re still overlaying `postLinkSignHook` in the bootstrap. We could presumably just drop the package entirely?

Done. I dropped it. It didn’t even cause rebuilds for me because it’s unused.

paparodeo commented:
> cc @NixOS/infra for bootstrap tools upload

adding @lovesegfault as per https://github.com/NixOS/nixpkgs/tree/master/maintainers/scripts/bootstrap-files for upload

emilazy commented:
AIUI the whole build infra team has access to do this now, or at least @Mic92? Not sure the exact details though.

@reckenrode Looks like there’s a merge conflict.

reckenrode commented:
@emilazy The merge conflict has been resolved.

lovesegfault commented:
I can upload these tomorrow when I'm on the computer :)

Mic92 commented:
Uploaded everything. Please merge if everything is fine.
